api/types/clustername.go

```diff
@@ -141,7 +141,9 @@ func (c *ClusterNameV2) CheckAndSetDefaults() error {
 	if c.Spec.ClusterName == "" {
 		return trace.BadParameter("cluster name is required")
 	}
-
+	if c.Spec.ClusterID == "" {
+		return trace.BadParameter("cluster ID is required")
+	}
```

Comment:
Just curious, is there a migration in v7 that sets this field if it's not set? Just to potentially prevent failures to start in case the resource w/o cluster ID was in the backend.

Reply:
Yep, that was performed by `migrateClusterID` that's also being removed with this PR.